hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ClusteringPlanActionExecutor.java

```diff
@@ -63,6 +63,7 @@ protected Option<HoodieClusteringPlan> createClusteringPlan() {
     int commitsSinceLastClustering = table.getActiveTimeline().getCommitsTimeline().filterCompletedInstants()
         .findInstantsAfter(lastClusteringInstant.map(HoodieInstant::getTimestamp).orElse("0"), Integer.MAX_VALUE)
         .countInstants();
+
     if (config.inlineClusteringEnabled() && config.getInlineClusterMaxCommits() > commitsSinceLastClustering) {
```

Comment:
This and the condition below guarantee that the clustering is only scheduled based on the max_commits config.  @eric9204 could you double check the logic?

Reply:
ok, so the issue we are trying to solve is:

there is a regular writer which just schedules clustering and we have a async clustering job which does the execution of clustering. 

if clustering is pending (may be will be executed by an async clustering job), every new successful commit with regular writer will keep adding new replacecommit.requested. 

If yes, then the fix makes sense to me. 
@yihua @danny0405 : wdyt.

Reply:
but one thing which I am finding it hard to comprehend is. wrt clustering, either both planning and execution is inline. or both are async atleast wrt spark datasource writer. So, not sure how the user ended up where clustering was just scheduled w/o getting to completion.

Reply:
Yes, the fix makes sense to me too.

>but one thing which I am finding it hard to comprehend

Flink writer schedules a clustering plan on each successful regular commit and there is a async pipeline that executes the clustering continuously, this patch can solve the problem that the clustering plan schedules too frequently if there is pending clustering.

So, +1 from my side.